### ISSUE TITLE
Inconsistency in External Link Opening Functionality 

### ISSUE THREAD

**Issue by vonProteus** [duckduckgo/iOS#3152]
Your app has become my go-to web browser. 
Unfortunately, I've noticed that external links some times are no longer opening properly again. 
I kindly request that you prioritize fixing this issue and, for the sake of consistency and reliability, consider implementing comprehensive end-to-end tests for it.

i have 7.129.1.0 version

https://github.com/user-attachments/assets/1fc83fd3-7639-410a-8281-0ce7e4456a87

**Comment by brindy** [comment]
Thanks for your report.  You're a couple of versions behind.  Can you please update and see if it's still a problem?

**Comment by vonProteus** [comment]
So far, in version 7.130.0.2, the issue hasn't occurred. However, it didn't happen earlier consistently either.
After a restart, the issue resolves itself temporarily for some time.

**Comment by vonProteus** [comment]
Previously, I encountered an issue where opening links from another app led to problems, as I mentioned earlier. However, I've noticed that this occurs less frequently now, and I no longer need to force-quit the app in order to open a link for a second time. Currently, my device is running version 7.131.0.1.

**Comment by vonProteus** [comment]
i localized another lesser instance of this error
 

https://github.com/user-attachments/assets/75180c50-6b8f-4727-8f3e-d4ef79659ddb


to reproduce it (in 7.131.0.1)
in `Setting` -> `Data Clearinga` ->  `Automatically Clear Data` -> set `Clear Tabs and Data` and `App Exit, Inactive for 5 Minutes`

then, I waited over five minutes before attempting to open an external link

**Comment by vonProteus** [comment]
it seems its back again (in 7.134.0.0)

**Comment by vonProteus** [comment]
looks like its is fixed in 7.142.1.1

**Comment by vonProteus** [comment]
sory its not fixed

**Comment by brindy** [comment]
Hi @vonProteus.  Thanks for keeping this up to date.  We've started a project to hopefully address this once and for all but it's going to be a few weeks before it is released, so watch this space.  We're tracking it here https://app.asana.com/0/414709148257752/1208360751695791/f

Since we're actively working on it I'm going to close this out, but I've added a note to ping here one the release is available.
